File: recursion/array-product-sum/javascript/ProductSum.js
function productSum(list, depth){
    let sum = 0;

    list.forEach(element => {
        if(Array.isArray(element))
        sum += productSum(element, depth + 1); 
        else
        sum += element;
    });

    return sum * depth;
}

const list = [1, 2, [2, 3], 6, [[2, 3], 1], 7];

console.log(productSum(list, 1));

File: recursion/array-product-sum/java/ProductSum.java
package com.basicsstrong.algorithm.recursion;

import java.util.List;

public class ProductSum {

	public static void main(String[] args) {
		
		List<Object> a1 = List.of(2,3);
		List<Object> a3 = List.of(a1, 1);
		List<Object> array = List.of(1, 2, a1, 6, a3, 7);
		
		System.out.println(productSum(array, 1));
		

	}
	
	static int productSum(List<Object> list, int depth) {
		
		int sum = 0;
		
		for (Object object : list) {
			if(object instanceof List) {
				sum += productSum((List<Object>)object, depth + 1);
			}else {
				sum += (int)object;
			}
		}
		
		return sum*depth;
		
	}

}
